extensions/coroutines-extensions/src/commonMain/kotlin/com/squareup/sqldelight/runtime/coroutines/FlowExtensions.kt

```diff
@@ -20,81 +20,67 @@ package com.squareup.sqldelight.runtime.coroutines
 
 import com.squareup.sqldelight.Query
 import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.FlowPreview
-import kotlinx.coroutines.channels.Channel.Factory.CONFLATED
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.channels.awaitClose
 import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.flowViaChannel
-import kotlinx.coroutines.flow.flowWith
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.flow.conflate
 import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.flow.mapNotNull
+import kotlinx.coroutines.withContext
 import kotlin.coroutines.CoroutineContext
 import kotlin.jvm.JvmName
 import kotlin.jvm.JvmOverloads
 
 /** Turns this [Query] into a [Flow] which emits whenever the underlying result set changes. */
-@FlowPreview
+@ExperimentalCoroutinesApi
 @JvmName("toFlow")
-fun <T : Any> Query<T>.asFlow(): Flow<Query<T>> = flowViaChannel(CONFLATED) { channel ->
+fun <T : Any> Query<T>.asFlow(): Flow<Query<T>> = callbackFlow<Query<T>> {
   val listener = object : Query.Listener {
     override fun queryResultsChanged() {
-      channel.offer(this@asFlow)
+      offer(this@asFlow)
     }
   }
   addListener(listener)
-  @Suppress("EXPERIMENTAL_API_USAGE") //TODO: Remove when invokeOnClose is no longer experimental, or use replacement.
-  channel.invokeOnClose {
+  awaitClose {
     removeListener(listener)
   }
-  channel.offer(this@asFlow)
-}
+  offer(this@asFlow)
```

Comment:
It seems like `offer(this@asFlow)` should be called before `awaitClose`. 
In my test case `Query` is not delivered until the underlying table is updated. and calling `offer` before `awaitClose` solved the problem.

Reply:
That's right, my mistake. That's probably what caused timeouts in tests. Solved, thank you.